waku/common/rate_limiter.go

```diff
@@ -186,6 +187,14 @@ func (r *PeerRateLimiter) Decorate(p RateLimiterPeer, rw p2p.MsgReadWriter, runL
 				// this relies on <nil> being the string representation of nil
 				// as IP() might return a nil value
 				ip = p.IP().String()
+
+				// If not the special <nil> case, take the hash of the IP address
+				// so it's not stored on our side
+				if ip != "<nil>" {
+					h := make([]byte, 64)
+					sha3.ShakeSum256(h, []byte(ip))
+					ip = fmt.Sprintf("%x", h)
+				}
```

Comment:
Could we just make this into a `obfuscateIp` helper?
```go
func obfuscateIp(ip string) string {
	if ip != "<nil>" {
		return ip
	}
	h := make([]byte, 64)
	sha3.ShakeSum256(h, []byte(ip))
	return fmt.Sprintf("%x", h)
}
```

Reply:
Yes, can I just call it `hashIPAddress` instead?

Reply:
whatever